### PR TITLE
Actions for Linux updates

### LINKED ISSUE #2451: [Bug] Better Linux CI actions
### Describe the issue

The Linux CI checks need some love.
For instance the `grep` is finding all lines with a 't' instead of all lines with a tab

It also seems the CI step is not working as intended since it does not report errors

### Steps to Reproduce

1. Open the PR

### Expected Behaviour

The CI step should detect tabs

### Save Game

Save game attached to report: Yes/No

### CorsixTH Version

0.66

### Operating System

Linux

### Theme Hospital Version

CD

### Gamelog.txt

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by tobylane**
*Fixes #2451*

**Describe what the proposed change does**
- Update tools
- Fix the check for tabs
- Spellcheck changes. To avoid needing to rebase all current PRs, it won't fail the run. I will remove this later
- Clang format's errors won't stop clang tidy from running
- Add a check that the windows config file is up to date

The new output is https://github.com/tobylane/CorsixTH/actions/runs/6840811854/job/18600472930#step:10:34

### CODE REVIEW

File: .github/workflows/Linux.yml [hunk 1]
```diff
@@ -101,35 +102,40 @@ jobs:
...
           cd build/CorsixTH
           ctest --extra-verbose --build-config Release --output-on-failure
           cd ../..
           # Build and package LevelEdit
           ant -buildfile LevelEdit/build.xml dist
-          # Run codespell
-          codespell --enable-colors --quiet-level 2 --skip="languages,corsix-th.6" \
-            -L sav,unexpect,persistance,defin,uint,inout,currenty,blong \
-            AnimView CorsixTH CorsixTH/Lua/languages/english.lua LevelEdit libs
+          # Run codespell (temporarily not a failure, Nov 2023)
```
Comment (nMustaki) [inline .github/workflows/Linux.yml:112]: Why not enable it ?

**Review by lewri — COMMENTED** [review]
Are we still required to test 5.3? Also are we specifically using >=5.4.2 or just latest?

File: .github/workflows/Linux.yml (hunk 1 above)
Comment (lewri) [reply .github/workflows/Linux.yml:112]: Temporary non-failure state to bypass some current PRs according to the pr summary. It'll be put back again later once the current stack of PRs is merged.

**Comment by Alberth289346** [comment]
It does much more than addressing updates for Linux actions. Best would be to make different PRs for different kinds of changes.

At least the `rnc.cpp` spell-fix is not quite feasible as far as I know. We agree it's a typo, but there are legal questions whether such a typo-fix is allowed by the license as it's modifying the license text itself.

**Comment by lewri** [comment]
@Alberth289346 can you review and if it's all good merge please? I can't merge using my resources because of the blocked 5.3 test.

**Comment by lewri** [comment]
@TheCycoONE if you could look at resolving this one when you can please =]

**Pull request merged**